src/test/groovy/com/commercehub/gradle/plugin/avro/OptionsFunctionalSpec.groovy

```diff
@@ -299,4 +305,50 @@ class OptionsFunctionalSpec extends FunctionalSpec {
         then:
         taskInfoAbsent || result.task(":generateAvroJava").outcome == SUCCESS
     }
+
+    @Unroll
+    def "supports configuration of dateTimeLogicalType to #dateTimeLogicalType"() {
+        given:
+        copyResource("user.avsc", avroDir)
+        buildFile << """
+        |avro {
+        |    dateTimeLogicalType = "${dateTimeLogicalType}"
+        |}
+        |""".stripMargin()
+
+        when:
+        def result = run("generateAvroJava")
+
+        then: "the task succeeds"
+        taskInfoAbsent || result.task(":generateAvroJava").outcome == SUCCESS
+        def content = projectFile("build/generated-main-avro-java/example/avro/User.java").text
+
+        and: "the specified dateTimeLogicalType is used"
+        content.contains("public void setBirthDate(${fieldClz} value)")
+
+        where:
+        dateTimeLogicalType                                                              | fieldClz
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JODA.name()                   | "org.joda.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JODA.name().toLowerCase()     | "org.joda.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JSR310.name()                 | "java.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.DEFAULT.name()                | actualDateTimeImplementationDefault
+        SpecificCompiler.DateTimeLogicalTypeImplementation.DEFAULT.name().toLowerCase()  | actualDateTimeImplementationDefault
```

Comment:
I did write the code such that the user can specify "DEFAULT" as a configuration value. I think this makes sense if the user intent is "use whatever upstream uses". However, see note below.